Source/NSGridView.m

```diff
@@ -0,0 +1,1280 @@
+      NSGridRow *row = [_rows objectAtIndex: i];
+
+      [row setGridView: self];
+      [c _setOwningRow: row];
+      [c _setOwningColumn: gc];
+      [v setFrame: f];
+      [c setContentView: v];
+      RELEASE(v);
+      [_cells insertObject: c
+                   atIndex: pos + i * [self numberOfColumns]];
+      RELEASE(c);
+      i++;
+    }
+  END_FOR_IN(views);
+
+  // Insert remaineder of cells for views not present..
+  NSUInteger r = [self numberOfColumns] - i;
+  NSUInteger idx = 0;
+  pos += i;
+  for(idx = 0; idx < r; idx++)
+    {
+      NSGridCell *c = [[NSGridCell alloc] init];
+      [_cells insertObject: c atIndex: pos + idx * [self numberOfColumns]];
+    }
+  
+  // Refresh...
+  [self _refreshCells];
+  return gc;
+}
+
+- (void) moveColumnAtIndex: (NSInteger)fromIndex toIndex: (NSInteger)toIndex
+{
+  NSArray *views = [self _viewsForColumnAtIndex: fromIndex];
+  [self removeColumnAtIndex: fromIndex];
+  [self insertColumnAtIndex: toIndex withViews: views];
+}
+
+- (void) removeColumnAtIndex: (NSInteger)index
+{
+  NSArray *objs = [self _cellsForColumnAtIndex: index];
+
+  [_columns removeObjectAtIndex: index];
+  [_cells removeObjectsInArray: objs];
+
+  [self _refreshCells];
+}
+
+- (NSGridCellPlacement) xPlacement
+{
+  return _xPlacement;
+}
+
+- (void) setXPlacement: (NSGridCellPlacement)x;
+{
+  _xPlacement = x;
+}
+
+- (NSGridCellPlacement) yPlacement;
+{
+  return _yPlacement;
+}
+
+- (void) setYPlacement: (NSGridCellPlacement)y;
+{
+  _yPlacement = y;
+}
+
+- (NSGridRowAlignment) rowAlignment;
+{
+  return _rowAlignment;
+}
+
+- (void) setRowAlignment: (NSGridRowAlignment)a;
+{
+  _rowAlignment = a;
+}
+
+- (CGFloat) rowSpacing
+{
+  return _rowSpacing;
+}
+
+- (void) setRowSpacing: (CGFloat)f
+{
+  _rowSpacing = f;
+}
+
+- (CGFloat) columnSpacing
+{
+  return _columnSpacing;
+}
+
+- (void) setColumnSpacing: (CGFloat)f
+{
+  _columnSpacing = f;
+}
+  
+- (void) mergeCellsInHorizontalRange: (NSRange)hRange verticalRange: (NSRange)vRange
+{
+  NSLog(@"Method %@ unimplemented.", NSStringFromSelector(_cmd));
+}
+
+// coding
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  [super encodeWithCoder: coder];
+  if ([coder allowsKeyedCoding])
+    {
+      [coder encodeInteger: _rowAlignment
+                    forKey: @"NSGrid_alignment"];
+      [coder encodeFloat: _columnSpacing
+                  forKey: @"NSGrid_columnSpacing"];
+      [coder encodeObject: _columns
+                   forKey: @"NSGrid_columns"];
+      [coder encodeFloat: _rowSpacing
+                  forKey: @"NSGrid_rowSpacing"];
+      [coder encodeObject: _rows
+                   forKey: @"NSGrid_rows"];
+      [coder encodeInteger: _xPlacement
+                    forKey: @"NSGrid_xPlacement"];
+      [coder encodeInteger: _yPlacement forKey: @"NSGrid_yPlacement"];
+      [coder encodeObject: _cells
+                   forKey: @"NSGrid_cells"];
+    }
+  else
+    {
+      [coder encodeValueOfObjCType:@encode(NSUInteger)
```

Comment:
Here and in the following lines spaces are missing.

Reply:
Corrected